internal/ioutil/ioutil.go

```diff
@@ -0,0 +1,45 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ioutil
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+// Ingest writes content into a temporary ingest file with the file name format
+// "oras_credstore_temp_{randomString}".
+func Ingest(dir string, content io.Reader) (path string, ingestErr error) {
+	tempFile, err := os.CreateTemp(dir, "oras_credstore_temp_*") // mode: 0600
+	if err != nil {
+		return "", fmt.Errorf("failed to create ingest file: %w", err)
+	}
+
+	path = tempFile.Name()
+	defer func() {
+		tempFile.Close()
```

Comment:
The error of `tempFile` is not checked.

Reply:
Is the error check necessary here? If tempFile failed to be closes, it would failed to be removed. But failing to remove should be ignorable?

Reply:
The `File.Close()` eventually calls [fclose()](https://www.man7.org/linux/man-pages/man3/fclose.3.html) or equivalent to flush the content to the disk.

If `io.Copy` returns nil error but `tempFile.Close()` returns an error, the actual file might be corrupted.

Reply:
Fixed.